packages/contracts-periphery/contracts/foundry-tests/Optimist.t.sol

```diff
@@ -0,0 +1,373 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/* Testing utilities */
+import { Test } from "forge-std/Test.sol";
+import { AttestationStation } from "../universal/op-nft/AttestationStation.sol";
+import { Optimist } from "../universal/op-nft/Optimist.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract Optimist_Initializer is Test {
+    address constant alice_admin = address(128);
+    address constant bob = address(256);
+    address constant sally = address(512);
+    string constant name = "Optimist name";
+    string constant symbol = "OPTIMISTSYMBOL";
+    string constant base_uri =
+        "https://storageapi.fleek.co/6442819a1b05-bucket/optimist-nft/attributes";
+    AttestationStation attestationStation;
+    Optimist optimist;
+
+    function _setUp() public {
+        // Give alice and bob and sally some ETH
+        vm.deal(alice_admin, 1 ether);
+        vm.deal(bob, 1 ether);
+        vm.deal(sally, 1 ether);
+
+        vm.label(alice_admin, "alice_admin");
+        vm.label(bob, "bob");
+        vm.label(sally, "sally");
+        _initializeContracts();
+    }
+
+    function _initializeContracts() internal {
+        attestationStation = new AttestationStation();
+        optimist = new Optimist(name, symbol, alice_admin, attestationStation);
+    }
+}
+
+contract OptimistTest is Optimist_Initializer {
+    function setUp() public {
+        super._setUp();
+        _initializeContracts();
+    }
+
+    function test_optimist_initialize() external {
+        // expect name to be set
+        assertEq(optimist.name(), name);
+        // expect symbol to be set
+        assertEq(optimist.symbol(), symbol);
+        // expect attestationStation to be set
+        assertEq(address(optimist.ATTESTATION_STATION()), address(attestationStation));
+    }
+
+    /**
+     * @dev Bob should be able to mint an NFT if he is allowlisted
+     * by the attestation station and has a balance of 0
+     */
+    function test_optimist_mint_happy_path() external {
+        // bob should start with 0 balance
+        assertEq(optimist.balanceOf(bob), 0);
+
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
```

Comment:
This is missing assertions of events

Reply:
An example of how to assert for calls and events being emitted:

https://github.com/ethereum-optimism/optimism/blob/d30e727f3116ab1d3e7226289f5238376c4a4fd7/packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol#L41